sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -184,4 +184,17 @@ abstract class ExternalCatalog {
 
   def listFunctions(db: String, pattern: String): Seq[String]
 
+  // --------------------------------------------------------------------------
+  // Resources
+  // --------------------------------------------------------------------------
+
+  /**
+   * Add a JAR resource to the underlying external catalog for DDL (e.g. CREATE TABLE) and DML
+   * (e.g., LOAD TABLE) operations.
+   *
+   * For example, when users create a Hive serde table, they can specify a custom
+   * Serializer-Deserializer (SerDe) class. When Hive metastore is unable to access the custom SerDe
+   * JAR (e.g., not on the Hive classpath), the JAR file must be added at runtime using this API.
+   */
+  def addJar(path: String): Unit
```

Comment:
After hive is consolidated into data source, custom serde should be the business of data source, not `ExternalCatalog`, right? And it is possibly that the external catalog is not hive metastore.

However, it might be too far to bring the consideration here.